### PR TITLE
Introduce new md inspection tool

### PR DESCRIPTION
**Pull request by dlsloan**
Displays superblock information and dumps data from the data region of
md array block devices. This is a great helper tool when debugging write
path data errors.

### CODE REVIEW

File: mdinspector.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    parser.add_argument('--start', type=int,
+                        help='Start dumping data region from starting offset')
+    parser.add_argument('--length', type=int, default=0,
+                        help='Maximum number of bytes to dump, 0 will read to end of disk')
+    args = parser.parse_args()
+
+    md_dev = MDBlkDev(args.dev)
+
+    if args.data:
+        data_pos = md_dev.sb_data_offset * 512
```
Comment (iomartin) [inline mdinspector.py]: `data_pos` gets overwritten below

File: mdinspector.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+        if args.start:
```
Comment (iomartin) [inline mdinspector.py]: You can just set the default for args.start to 0 and then you don't need the if

File: mdinspector.py (hunk 1 above)
Comment (dlsloan) [reply mdinspector.py]: oops, that's a peice of old code lingering

File: mdinspector.py (hunk 2 above)
Comment (dlsloan) [reply mdinspector.py]: Similar to the other comment, old code that needs more clean up, fixed now

**Comment by lsgunth** [comment]
I don't understand how this helps at all? All my debugging I've never needed to parse the super block.

**Comment by dlsloan** [comment]
It looks up the start of data and tells you which disk is which, those are the two most useful bits

**Review by lsgunth — APPROVED** [review]
Ah, I guess I just always made assumptions or calcualtions about the disk offsets...

Can we just put this in a debug directory or something like that? I have some other stuff I can add to it.

**Pull request merged**